### ISSUE TITLE
toType conversion logic incorrect

### ISSUE THREAD

**Issue by mwohlan** [unjs/undio#7]
### Environment

node

### Reproduction

see description

### Describe the bug

In the source code you can see the helper functions not using the detectType function to detect the "fromType", leading to conversion errors.

```ts
function convertTo(toType, input) {
  const _map = _to[toType];
  if (!_map) {
    throw new Error(`Conversion to ${toType} is not supported.`);
  }
  const fromType = detectType(input);
  return _convertTo(fromType, input, _map);
}
const toArrayBuffer = (input) => _convertTo("ArrayBuffer", input, _toArrayBuffer);
// should be (input) => _convertTo(detectType(input), input, _toArrayBuffer);
// same applies fot the other helper functions
const toBlob = (input) => _convertTo("Blob", input, _toBlob);
const toDataView = (input) => _convertTo("DataView", input, _toDataView);
const toNumberArray = (input) => _convertTo("NumberArray", input, _toNumberArray);
const toReadableStream = (input) => _convertTo("ReadableStream", input, _toReadableStream);
const toResponse = (input) => _convertTo("Response", input, _toResponse);
const toString = (input) => _convertTo("String", input, _toString);
const toUnit8Array = (input) => _convertTo("Uint8Array", input, _toUint8Array);
function _convertTo(name, input, map) {
  const converter = map[name];
  if (converter === void 0) {
    throw new Error(`Conversion from ${name} is not supported.`);
  }
  return converter(input);
}
```

**Comment by mwohlan** [comment]
Sorry, I should have done it right away:

https://stackblitz.com/edit/github-1n29rg?file=server%2Froutes%2Findex.ts

**Comment by itpropro** [comment]
This is actually a bug @pi0. The API of `convertTo` and `_convertTo` is not used the same way and `_convertTo` is not covered by tests currently.
I added tests to also cover the `to{TYPE}` functions and fixed the bug in #9, looking forward to your feedback!

**Comment by pi0** [comment]
Should be fixed in next release via #9
